astropy/io/fits/header.py

```diff
@@ -1069,10 +1069,10 @@ def update_from_dict(k, v):
                 card = Card(*((k,) + v))
             else:
                 raise ValueError(
-                    "Header update value for key %r is invalid; the "
+                    f"Header update value for key {k!r} is invalid; the "
```

Comment:
Just curious, why is this not just ``{k}``?

Reply:
`!r` calls `repr` instead of `str`, which is sometimes desirable and in particular will put strings in quotes. Here it's a strictly equivalent refactor.

Reply:
I'm happy to change it, I was just keeping this strictly translated. But yes, `{k}` is equivalent.
IMO I don't sweat these (unless for line length) and I'm just waiting for `ruff` to eventually systematize in one direction or another and auto fix everything with a pre-commit autoupdate. IDK why they haven't yet. A quick google found no discussion of this. Perhaps I'm missing a subtlety where `f"{k}"` and `f"{k!r}"` aren't actually equivalent. https://docs.astral.sh/ruff/rules/explicit-f-string-type-conversion/#why-is-this-bad talks about custom `__format__`, so maybe?

Reply:
> Perhaps I'm missing a subtlety where `f"{k}"` and `f"{k!r}"` aren't actually equivalent.

They only appear equivalent for types that implement either `__str__` or `__repr__` but **not** both (because they default to each other). They are not the same in the `str` type.

Reply:
Have you looked up any explicit examples?
```python
>>> greeting = "Hello!"
>>> print(f"{greeting}")
Hello!
>>> print(f"{greeting!r}")
'Hello!'
```

Reply:
This PR is meant to refactor code, not change behavior, so we should stick with`{k!r}`.